src/natcap/invest/seasonal_water_yield/seasonal_water_yield.py

```diff
@@ -749,8 +755,28 @@ def execute(args):
     output_align_list = [
         file_registry['lulc_aligned_path'], file_registry['dem_aligned_path']]
     if not args['user_defined_local_recharge']:
-        precip_path_list = _get_monthly_file_lists(N_MONTHS, args['precip_dir'])
-        et0_path_list = _get_monthly_file_lists(N_MONTHS, args['et0_dir'])
+        month_indexes = [m+1 for m in range(N_MONTHS)]
+
+        precip_df = MODEL_SPEC.get_input(
+            'precip_raster_table').get_validated_dataframe(
+            args['precip_raster_table'])
+        precip_df.sort_values('month', inplace=True)
+        if not month_indexes == precip_df['month'].tolist():
+            raise ValueError(
+                'Precipitation raster table must include the '
+                'month index values 1-12.')
+
+        et0_df = MODEL_SPEC.get_input(
+            'et0_raster_table').get_validated_dataframe(
+            args['et0_raster_table'])
+        et0_df.sort_values('month', inplace=True)
+        if not month_indexes == et0_df['month'].tolist():
+            raise ValueError(
+                'ET0 raster table must include the '
```

Comment:
Can we phrase this to cover all conditions under which the check won't pass? Maybe something like 'ET0 table "month" column must include exactly the values 1 - 12 and no other values'